### PR TITLE
Fix worst pylint offenders

### PR DESCRIPTION
**Pull request by ericvaandering**
I'm going through trying to fix the "red" errors that pylint is reporting

### CODE REVIEW

**Comment by ericvaandering** [comment]
@ticoann please review and merge.

**Comment by amaltaro** [comment]
@ericvaandering how about you squash all these changes?

File: src/python/WMComponent/AnalyticsDataCollector/Plugins/Tier0Plugin.py
Comment (amaltaro) [inline src/python/WMComponent/AnalyticsDataCollector/Plugins/Tier0Plugin.py]: Good one!

File: src/python/WMComponent/JobCreator/JobCreatorSiteDBInterface.py
Comment (amaltaro) [inline src/python/WMComponent/JobCreator/JobCreatorSiteDBInterface.py]: I'll actually check this code out, I don't think it's used anywhere

File: src/python/WMComponent/JobCreator/JobCreatorWorker.py
Comment (amaltaro) [inline src/python/WMComponent/JobCreator/JobCreatorWorker.py]: Hmm, I guess it was never used then(?)

**Review by amaltaro — APPROVED** [review]
Looks good to me. However, how about we wait for the testbed validation phase to be over before it gets merged.

**Comment by ericvaandering** [comment]
I figured I would wait until it was about to be merged to squash things. Makes life easier in case something has to be backed out.

File: src/python/WMComponent/JobCreator/JobCreatorWorker.py
Comment (ericvaandering) [reply src/python/WMComponent/JobCreator/JobCreatorWorker.py]: Quite possibly. Some things I tried to find out if they were used or not. Others I just tried to fix up

File: src/python/WMComponent/AnalyticsDataCollector/Plugins/Tier0Plugin.py
Comment (ericvaandering) [reply src/python/WMComponent/AnalyticsDataCollector/Plugins/Tier0Plugin.py]: Took me a minute to understand what pylint was telling me which is that the 2nd exception was a subclass of the first, ergo never gets run.

**Comment by amaltaro** [comment]
@ticoann I think we can merge this one as well (after commits are squashed)

**Comment by amaltaro** [comment]
test this please

**Comment by ericvaandering** [comment]
Squashed

**Pull request merged**
